catalog/dags/data_refresh/dag_factory.py

```diff
@@ -261,11 +263,31 @@ def create_data_refresh_dag(data_refresh: DataRefresh, external_dag_ids: Sequenc
             },
         )
 
+        index_suffix = XCOM_PULL_TEMPLATE.format(
+            data_refresh_group.get_child_by_label("generate_index_suffix").task_id,
+            "return_value",
+        )
+        trigger_filtered_index_creation = TriggerDagRunOperator(
+            task_id=f"trigger_create_filtered_{data_refresh.media_type}_index",
+            trigger_dag_id=f"create_filtered_{data_refresh.media_type}_index",
+            params={
```

Comment:
This needs to be [`conf`](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/operators/trigger_dagrun/index.html#airflow.operators.trigger_dagrun.TriggerDagRunOperator), otherwise `params` usage here will be interpreted as [task-level params](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/params.html#task-level-params). I tried running the DAGs locally and the triggered create filtered index DAG failed because this did not appropriately pass `force` and the downstream DAG started before the upstream DAG was complete.

```suggestion
            conf={
```

Reply:
I don't understand :thinking: When I try this locally it does work, the data refresh is able to trigger the filtered index creation, i.e., the params are passed to the DAG.

Can you share more specifics about how you were able to reproduce the issue? The documentation you linked was what I referenced to use the params there, so I'm confused what to understand about it that should indicate the usage of conf instead :sweat:

Reply:
Just because the DAG itself gets triggered doesn't necessarily mean that the configuration parameters defined here are sent to the DAG. In my case, the DAG was started but did not receive the configuration parameters. I'm also realizing now that the `params=` you define for the DAG is also incorrect, since those are [DAG-level params](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/params.html#dag-level-params).

To rephrase the two different (but confusingly named Airflow concepts here):
- [**Parameters**](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/params.html) are a way to add variables _into the jinja templating that Airflow tasks use to render their inputs_. These are accessed via their names in templates, e.g. a Param named `x` is going to be accessible via `{{ params.x }}`. 
- [**DAG run configuration**](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/dag-run.html#passing-parameters-when-triggering-dags) are the inputs that you can provide _to a specific DAG run_. These are accessible via `dag_run.conf`, e.g. `{{ dag_run.conf["x"] }}`. 

The [template reference](https://airflow.apache.org/docs/apache-airflow/stable/templates-ref.html#variables) is useful here, although it _looks_ like we might also be able to use `params` for DAG run configuration if we set the `core.dag_run_conf_overrides_params` system configuration 😮

We use `dag_run.conf` as part of the ingestion DAGs: https://github.com/WordPress/openverse/blob/aedc9c16ce5ed11709e2b6f0b42bad77ea1eb19b/catalog/dags/providers/factory_utils.py#L32

This does make me wonder, I will need to try to see if enabling that setting makes this behave as you're expecting, the docs make it seem as simple as that: https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#dag-run-conf-overrides-params. If we can get that to work I think I'd prefer it, especially since `Params` have a lot more configuration options!

Reply:
My understanding of the above evolved as I was writing that comment ☝🏼 I need to head out for the day but the TL;DR is that I think everything should be right based on how you've defined it but maybe I have `AIRFLOW__CORE__DAG_RUN_CONF_OVERRIDES_PARAMS` not set correctly locally or something (or we may need to define it explicitly, even though it's supposed to default to `True`? 🤔)

Reply:
Oh one other thing, the index UUIDs matched up when you ran this locally? For me they did not, another indication that the DAG run config/params weren't working:
![Screenshot_2023-04-20_16-31-13](https://user-images.githubusercontent.com/10214785/233519989-3bba085c-9c16-4606-a972-557a02ddf213.png)

Reply:
I don't think the DAG configuration is incorrect. Your description of the usage of it precisely matches what I intended: they're DAG level configuration variables that are passed by the DAG to the individual tasks. When you trigger the DAG "w/ config" from the UI and pass these variables they get correctly propogated through the tasks. If you pass `destination_index_suffix` as `"some-other-suffix"` it does work, at least when I run it locally.

I think you are correct about them not being passed. They do need to be passed with `conf`, but the explanation you gave didn't make sense to me. The `params` I've passed to the trigger task are passed to the task, `conf` is passed by the trigger task on to the DAG that it triggers. I didn't read the trigger task's documentation carefully. So you're right there.

Also: I'm not sure how but I must have done something terribly wrong when testing the data refresh DAG locally as I swear I was able to get it to work but now it does not, as you described.

I'll update this to use conf, and verify, but I don't believe there are any other changes needed.

Reply:
Also: I realised we probably want a sensor to wait for the filtered index creation to finish, potentially? :thinking: What do you think?

Reply:
Fixed the param passing in https://github.com/WordPress/openverse/pull/1833/commits/541648466e405bde9c563132ecde8bbc80b2c623

I'll update the testing instructions to share how you can prove it worked.

Reply:
Ah, okay I see what you mean. I think the `params` construct has evolved since I used it last, it used to be more limited in scope. Apologies for my misinterpretation there!

As for a sensor, I think it would be necessary if there are any cases where other operations would need to wait for the filtered index creation to be complete. But isn't that what the `trigger_and_wait_for_task` function does?

Reply:
Sorry, just to clarify, the sensor I was wondering about would be added to the Data Refresh DAG to wait for the filtered index creation DAG to finish. We'd only need it if we wanted the Data Refresh DAG's full timeline to include the length of the filtered index creation DAG that it triggers. Right now it triggers the DAG and then finishes immediately. I don't know what is more useful from an observability perspective, is all.

Reply:
Ohhh, I see, thanks for clarifying! I think having that final sensor would make sense, since the data refresh is technically "still running" when the filtered index creation is happening, right?